src/pages/game/guess-bar/guessSlice.ts

```diff
@@ -1,23 +1,51 @@
 import { createSlice, PayloadAction } from '@reduxjs/toolkit';
 
 import { RootState } from '../../../app/store';
+import { guessStateType } from '../../../types/guessStateType';
+import { POLISH_COMMON_WORDS } from '../../../utils/constants';
 
-const initialState = {
-    text: '',
+const initialState: guessStateType = {
+    guessedWords: [],
+    guessHistory: [],
+    isAlreadyGuessed: false,
 };
 
 export const guessSlice = createSlice({
     name: 'guess',
     initialState,
     reducers: {
         setGuessText: (state, action: PayloadAction<string>) => {
-            state.text = action.payload;
+            const guessedWord = action.payload;
+            const updatedGuessWordsArray = [...state.guessedWords, guessedWord];
+
+            if (state.guessedWords.length === 0) {
+                return (state = {
```

Comment:
is "state = " needed here? cant you simply return? the same applies to line 33 ;)

Reply:
done.